apm-sniffer/optional-plugins/trace-ignore-plugin/src/main/java/org/apache/skywalking/apm/plugin/trace/ignore/matcher/FastPathMatcher.java

```diff
@@ -124,7 +124,7 @@ private boolean wildcardMatch(String pat, int p, String str, int s) {
     private boolean multiWildcardMatch(String pat, int p, String str, int s) {
         // End of pattern, just check the end of string is '/' quickly.
         if (p >= pat.length() && s < str.length()) {
-            return str.charAt(str.length() - 1) != '/';
+            return true;
         }
```

Comment:
Maybe the comment need be changed.

```Java
        // End of pattern, make matching success quickly.
        if (p >= pat.length() && s < str.length()) {
            return true;
        }
```

Reply:
In my originally design, the `/eureka/**` not match the `/eureka/client/` is a feature, but I have no preference for this, I can accept it regardless of whether it matches or does not match.

Could you show your use cases and benefits for it?

Reply:
I think because some users think all things under `/eureka/` belong to this match rule.

Reply:
@devkanro Could you share a little more about why `/eureka/client/` doesn't belong to `/eureka/**`? I hope we could have a fully evaluation considering this was being asked more than once.(This is the first fix)

Reply:
Ummm...I can't remember what I thought at the time...

Maybe it's because `/eureka/` doesn't match `/eureka`, I want to make it same as `/eureka/**` case?

I have changed my mind, the `/eureka/client/` should belong to `/eureka/**`.

LGTM

Reply:
And maybe the rule of `wildcardMatch` also should be changed?
Should `/eureka/*` match with `/eureka/client/` ?

Reply:
>/eureka/ doesn't match /eureka

I think they should be same, do you know any different in some use cases?
I feel they are same. 

> And maybe the rule of wildcardMatch also should be changed?
Should /eureka/* match with /eureka/client/ ?


Yes, it should be supported, too.

Reply:
what is final dicussion?

Reply:
The current resolution is, what your proposal is correct. Just follow the polish requirements.

Reply:
@wu-sheng 
> /eureka/ doesn't match /eureka
Ummm, The reason I do this is to simplify the state machine. There are no other special considerations.